src/utils/utils.js

```diff
@@ -3,3 +3,59 @@ export function isStateless(component) {
     // `() => {}` via Babel has prototype too.
     return !(component.prototype && component.prototype.render)
 }
+
+function getMixins(target, methodName) {
+    const mixins = (target["__$mobxMixins"] = target["__$mobxMixins"] || {})
+    const methodMixins = (mixins[methodName] = mixins[methodName] || {})
+    methodMixins.pre = methodMixins.pre || []
+    methodMixins.post = methodMixins.post || []
+    return methodMixins
+}
+
+export function patch(target, methodName, mixinMethod, runMixinFirst = false) {
+    const mixins = getMixins(target, methodName)
+
+    if (runMixinFirst) {
+        mixins.pre.unshift(mixinMethod)
+    } else {
+        mixins.post.push(mixinMethod)
+    }
+
+    let realMethod = target[methodName]
+    if (typeof realMethod === "function" && realMethod["__$mobxMixin"]) {
+        // already patched, do not repatch
+        return
+    }
+
+    function getFunction(...args) {
+        const mixins = getMixins(this, methodName)
+
+        mixins.pre.forEach(pre => {
+            pre.apply(this, args)
+        })
+
+        if (realMethod !== undefined && realMethod !== null) {
+            realMethod.apply(this, args)
+        }
+
+        mixins.post.forEach(post => {
+            post.apply(this, args)
+        })
+    }
+    getFunction["__$mobxMixin"] = true
+
+    const newDefinition = {
```

Comment:
Optimization: performance / memory consumption is significant lower if property descriptors are cached and reused per name. To achieve that `getFunction` needs to be generic, for that, `realMethod` needs not to be used. That could be done by for example storing the real method on the `mixins.real`  property